openml-lightgbm/lightgbm-provider/src/main/java/com/feedzai/openml/provider/lightgbm/LightGBMSWIG.java

```diff
@@ -144,7 +144,42 @@ private void copyDataToSWIGInstance(final Instance instance) {
         }
     }
 
+    /**
+     * Returns the features contributions for the current instance.
+     *
+     * @param instance The instance from pulse.
+     * @return double[] array with features contributions.
+     * @since 1.2.2
+     */
+    public double[] getFeaturesContributions(final Instance instance) {
 
+        // we need to lock the resources to avoid having multiple threads sharing the same swig resources.
+        synchronized (this.swigResources) {
+            final double[] contributions = new double[this.schemaNumFields];
+
+            copyDataToSWIGInstance(instance);
+            final int returnCodeLGBM = lightgbmlibJNI.LGBM_BoosterPredictForMatSingleRowFast(
+                    this.swigResources.swigFastConfigContributionsHandle,
+                    this.swigResources.swigInstancePtr,
+                    this.swigResources.swigOutLengthInt64Ptr,
+                    this.swigResources.swigOutContributionsPtr // preallocated memory
+            );
+
+            if (returnCodeLGBM == -1) {
+                throw new LightGBMException();
+            }
+
+            for (int index = 0; index < this.schemaNumFields; ++index) {
+                contributions[index] = lightgbmlibJNI.doubleArray_getitem(
+                        this.swigResources.swigOutContributionsPtr,
+                        index
+                );
+            }
```

Comment:
Nitpick: maybe change this to something more functional?

Reply:
The reason that I kept this approach is that I'm using array with primitive value double. Based on section [Lambda vs Traditional](https://betterprogramming.pub/how-to-iterate-with-java-56b0fd83bbfc), the advantage of using lambda will be if we want to parallelize this iteration.
I will create a unit test in a separate branch to see if parallelizing this iteration we can have same results, if so I'll change to more functional approach.

Reply:
There's probably no need to parallelize something as trivial as this. Or could the the JNI access be slow?
Anyway, I was just suggesting for aesthetical/convention purposes. We tend to use the functional/streams approach in our codebases, so it is nice to always follow the same